### ISSUE TITLE
[A11y_Bug]: Search elements at different regions cannot have the same label

### ISSUE THREAD

**Issue by gcattan** [IBMa/equal-access#1582]
### Project

a11y checker

### Browser

Chrome

### Operating System

Windows

### Automated testing tool and ruleset

IMB Equal Accessibility Checker

### Description

We have a html page with 2 search bars: one on top of a grid, and another on top of a flyover (which can be opened by clicking on a contextual action). 

When the flyover is opened, there is the following accessibility issue:

![image](https://github.com/IBMa/equal-access/assets/6229031/c10414bb-f872-4c2f-ab5c-7fa514b46515)

I think in this case, it is fine to have the same label for the two elements with role `search`, because they are under different "region". 
The title of the grid and the flyover are different, and I believe it is clear that the label refers to different contexts.

### Steps to reproduce

- Add two elements with a "search" role and the same label. 
- An accessibility error is raised, even if they are located in differents regions.

**Comment by tombrunet** [comment]
Simply being in different regions does not make them unique. The label is "Search", but Search what? Screen reader users don't necessarily have any indication of flyouts and such, they're just two search regions on the page with no disambiguation as to which is which.

**Comment by gcattan** [comment]
I disagree. I have an example with a Search integrated to a page, and another Search inside a flyover on top of the page.
Because the page and the flyover have different titles, it is clear enough in this context to what "Search" refers to.

**Comment by philljenkins** [comment]
@gcattan 
1. ARIA Authoring Practices guide says:
> Search
> A [search](https://w3c.github.io/aria/#search) landmark contains a collection of items and objects that, as a whole, combine to create search functionality.
> 
> Use the search landmark instead of the form landmark when the form is used for search functionality.
If a page includes more than one search landmark, each should have a unique label (see [Step 3](https://www.w3.org/WAI/ARIA/apg/practices/landmark-regions/#aria_lh_step3) above).

2. It is not as obvious to the screen reader or magnifier user when the label is not unique, It requires the user extra steps to find and remember (create a mental model) the title and/or heading.  JAWS and other assistive technology will announce when content is changing from one region to another, that is NOT the case when changing content between headings.

**Comment by gcattan** [comment]
@philljenkins yes. I agree, but this is not exactly what I asked. Let me try to illustrate my use cases with Word for example.

- You have two pages. Both pages have one unique search, with a title that is different. The pages are embedded into a single screen or on two different screens - that depends on the size of the screen for example. I guess that when the two pages are rendered separately, there are no accessibility errors, but it could be one if there are rendered in together. 

![image](https://github.com/IBMa/equal-access/assets/6229031/e8a7cfe0-f083-4adc-87f7-91531930f155)

- You have one page with one heading and a unique search. But then you have a flyover/popup/modal with also a search functionality. This flyover/popup/modal can be displayed over any page, which one can or not include a search box. This flyover/popup/modal could also be displayed on another screen depending on the size of the device. When this flyover is displayed over a page that also includes a search, there is an error :
![image](https://github.com/IBMa/equal-access/assets/6229031/13d4936a-073b-414c-83b0-75f6579f51b8)


In my opinion, these are valid use cases where you do not need to have search boxes with different labels.

**Comment by philljenkins** [comment]
I believe using [aria-labelledby=](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA16) is the exact purpose of programmatically associating the "Page x" text with the appropriate "Search".  That the title, label, or text is in a unique hierarchy is not sufficient.

**Comment by gcattan** [comment]
Ok, I see. Thank you for your recommendation. 
Just a side question, but is it the same for example in Android/iOs devices? If you have two fragments of a same activity, rendered or not on the same screen, depending on the size of the device, with each one a search, should you be also using aria-labelledby?

**Comment by mbgower** [comment]
@gcattan since there was a query today in a related issue, this got brought to my attention.

In a nutshell, there are a few considerations:
1. Content in a popover or other dynamic chunk of information is likely on the same page as its trigger. If the popover content is hidden from all users until open, you shouldn't get an error running the checker -- unless you rerun the checker when it's open.
2. In the context of having just opened a dynamic piece of content and being prompted to "search", I take your point that a user should understand the context is different and this is not the same search mechanism they previously encountered.
3. But it would be very unusual, in my experience, for the scope of the search to be the same. So I'm inclined to say that a good design would provide that context to all users with a unique label/placeholder. Where this context is provided by the title of the page, that's where it makes sense to pick up that value with aria-labelledby.
4. In the example you provided, it seems clear to me that the search on page 1 or 2 is likely to be a different context than the popover search, which seems to be a Find files function (or possibly something refined with the "Plus" dropdown). If the designer chooses not to make the distinction at the placeholder level, then I think it falls to the remediator to figure out what context _is_ provided visually and attempt to reinforce programmatically.

**Comment by gcattan** [comment]
Hi @mbgower , thank you for your input.

> In the context of having just opened a dynamic piece of content and being prompted to "search", I take your point that a user should understand the context is different and this is not the same search mechanism they previously encountered.

As I remember, it was the case. The popover looked a bit as a side panel that was opened briefly for some configuration and then closed. But I am not working on the product anymore. Tagging @fbarroso24 for confirmation.
